### PR TITLE
fix airgapped cluster e2e image pull

### PR DESCRIPTION
**Pull request by fseldow**
**What type of PR is this?**

/kind test

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

File: e2e/config/config.go
Comment (cameronmeissner) [inline e2e/config/config.go:18]: im assuming this is just for testing atm

Comment (fseldow) [reply e2e/config/config.go:18]: the name is for temp to avoid conflict with old tests

File: e2e/config/config.go
Comment (AlisonB319) [inline e2e/config/config.go:16]: We will want to revert these changes when merging the PR. Then once this PR is merged, delete our e2e resource group and re-run an airgap test and the resources will get recreated.

File: e2e/kube.go
Comment (AlisonB319) [inline e2e/kube.go:96]: Why is this able to be reverted? 

If so, can you remove the isAirgap variable from the function since it isn't being used.

File: e2e/cluster.go
Comment (AlisonB319) [inline e2e/cluster.go:74]: Are you sure we want every e2e test running with the network plugin? I just want to make sure we are running the e2e's as close as possible to what a customer would be running without assuming they have anything installed

Since I'm assuming Test_Ubuntu2204Gen2ContainerdAirgapped_K8sNotCached needs the plugin to pass, maybe just that test should use the AzureNetwork cluster?

Comment (fseldow) [reply e2e/cluster.go:74]: in fact we do not support kubenet cluster for network isolated cluster. So that it should be azure network cluster. However, my latest test fails with node not ready. Still restricting the test range. Maybe in this pr we may remain kubenet and unblock the test first

File: e2e/config/config.go
Comment (fseldow) [reply e2e/config/config.go:18]: mark it as temp, since i need to ensure the flow can work with different name

File: e2e/kube.go
Comment (fseldow) [reply e2e/kube.go:96]: it is to rever the prefix `aks` in this pr

Comment (AlisonB319) [reply e2e/kube.go:96]: I'm not sure I quite understand why removing the aks prefix means we don't need to be able to pull cbl-mariner using our ACR? This might be why you are seeing node not ready?

File: e2e/cluster.go
Comment (AlisonB319) [reply e2e/cluster.go:74]: I just ran `Test_Ubuntu2204Gen2ContainerdAirgapped_K8sNotCached` on local (with a test rg and acr) and it was able to pass. Are you still seeing node not ready?

File: e2e/kube.go
Comment (AlisonB319) [reply e2e/kube.go:96]: Oh my bad, when I left this comment I was using a vscode extension that showed a different diff of the entire isAirgap statement removed.

File: e2e/cluster.go
Comment (AlisonB319) [reply e2e/cluster.go:74]: For what its worth, the kubelet logs show 
```
Nov 26 09:14:11 abe2e-2024-11-26-bj4f-ubuntu2204gen2containerdairgappedk8000000 kubelet[3491]: I1126 09:14:11.829439    3491 csi_plugin.go:99] kubernetes.io/csi: Trying to validate a new CSI Driver with name: disk.csi.azure.com endpoint: /var/lib/kubelet/plugins/disk.csi.azure.com/csi.sock versions: 1.0.0
Nov 26 09:14:11 abe2e-2024-11-26-bj4f-ubuntu2204gen2containerdairgappedk8000000 kubelet[3491]: I1126 09:14:11.829463    3491 csi_plugin.go:112] kubernetes.io/csi: Register new plugin with name: disk.csi.azure.com at endpoint: /var/lib/kubelet/plugins/disk.csi.azure.com/csi.sock
Nov 26 09:15:32 abe2e-2024-11-26-bj4f-ubuntu2204gen2containerdairgappedk8000000 kubelet[3491]: E1126 09:15:32.347559    3491 kubelet_node_status.go:456] "Node not becoming ready in time after startup"
Nov 26 09:15:32 abe2e-2024-11-26-bj4f-ubuntu2204gen2containerdairgappedk8000000 kubelet[3491]: E1126 09:15:32.839128    3491 kubelet.go:2892] "Container runtime network not ready" networkReady="NetworkReady=false reason:NetworkPluginNotReady message:Network plugin returns error: cni plugin not initialized"
```
for why node is not ready when using getAzureNetworkClusterModel

**Review by AlisonB319 — APPROVED** [review]

**Pull request merged**
